1.21.3/src/main/java/dev/u9g/minecraftdatagenerator/generators/BiomesDataGenerator.java

```diff
@@ -0,0 +1,115 @@
+package dev.u9g.minecraftdatagenerator.generators;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import dev.u9g.minecraftdatagenerator.util.DGU;
+import net.minecraft.registry.DynamicRegistryManager;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.tag.BiomeTags;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.biome.Biome;
+
+public class BiomesDataGenerator implements IDataGenerator {
+    private static String guessBiomeDimensionFromCategory(Biome biome) {
+        var biomeRegistry = DGU.getWorld().getRegistryManager().getOrThrow(RegistryKeys.BIOME);
+        if (biomeRegistry.getEntry(biome).isIn(BiomeTags.IS_NETHER)) {
+            return "nether";
+        } else if (biomeRegistry.getEntry(biome).isIn(BiomeTags.IS_END)) {
+            return "end";
+        } else {
+            return "overworld";
+        }
+    }
+
+    private static String guessCategoryBasedOnName(String name, String dimension) {
+        if (dimension.equals("nether")) {
+            return "nether";
+        } else if (dimension.equals("end")) {
+            return "the_end";
+        }
+
+        if (name.contains("end")) {
+            System.out.println();
+        }
```

Comment:
You sure this should be left in? looks like debug

Reply:
Gave you contributor access @AlexProgrammerDE so you can fix that kind of stuff directly